src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToArrayTransformer.php

```diff
@@ -167,12 +167,12 @@ public function reverseTransform($value)
         try {
             $dateTime = new \DateTime(sprintf(
                 '%s-%s-%s %s:%s:%s',
-                empty($value['year']) ? $this->referenceDate->format('Y') : $value['year'],
-                empty($value['month']) ? $this->referenceDate->format('m') : $value['month'],
-                empty($value['day']) ? $this->referenceDate->format('d') : $value['day'],
-                empty($value['hour']) ? $this->referenceDate->format('H') : $value['hour'],
-                empty($value['minute']) ? $this->referenceDate->format('i') : $value['minute'],
-                empty($value['second']) ? $this->referenceDate->format('s') : $value['second']
+                !isset($value['year']) ? $this->referenceDate->format('Y') : $value['year'],
```

Comment:
instead, use null coalescing operator everywhere
```suggestion
                $value['year'] ?? $this->referenceDate->format('Y'),
```

Reply:
Ah, was thinking about that but wasn't sure anymore when they did introduce that but seems it was in 7.0?